### PR TITLE
Upgrade/800/server profile

### PR DESCRIPTION
**Pull request by sijeesh**
### Description
Upgraded server profile to support OneView API version 800

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for Python 2.7+ & 3.4+(`$ tox`).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: endpoints-support.md [hunk 1]
```diff
@@ -471,23 +471,23 @@
...
-|<sub>/rest/server-profiles</sub>                                                         | POST     | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:
-|<sub>/rest/server-profiles</sub>                                                         | DELETE   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:
-|<sub>/rest/server-profiles/available-networks</sub>                                      | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:
-|<sub>/rest/server-profiles/available-servers</sub>                                       | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:
-|<sub>/rest/server-profiles/available-storage-system</sub>                                | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:
-|<sub>/rest/server-profiles/available-storage-systems</sub>                               | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:
```
[the remark this answers is not in the document]
Comment (sijeesh) [reply endpoints-support.md]: will remove it.

File: hpOneView/resources/servers/server_profiles.py [hunk 2]
```diff
@@ -333,8 +229,8 @@ def get_available_networks(self, **kwargs):
         Returns:
             list: Available networks.
         """
-        uri = self.__build_uri_with_query_string(kwargs, '/available-networks')
-        return self._client.get(uri)
+        uri = self._helper.build_uri_with_query_string(kwargs, '/available-networks')
+        return self._helper.do_get(uri)
 
     def get_available_servers(self, **kwargs):
         """
```
Comment (soodpr) [inline hpOneView/resources/servers/server_profiles.py]: we need to add the filter query param in the comments section

File: hpOneView/resources/servers/server_profiles.py [hunk 3]
```diff
@@ -399,8 +295,8 @@ def get_available_storage_systems(self, start=0, count=-1, filter='', sort='', *
         Returns:
             list: Available storage systems.
         """
-        uri = self.__build_uri_with_query_string(kwargs, '/available-storage-systems')
-        return self._client.get_all(start=start, count=count, filter=filter, sort=sort, uri=uri)
+        uri = self._helper.build_uri_with_query_string(kwargs, '/available-storage-systems')
+        return self._helper.get_all(start=start, count=count, filter=filter, sort=sort, uri=uri)
 
     def get_available_targets(self, **kwargs):
         """
```
Comment (soodpr) [inline hpOneView/resources/servers/server_profiles.py]: we need to add the filter query param in the comments

File: examples/server_profiles.py [hunk 4]
```diff
@@ -24,234 +24,216 @@
...
+
+# Get the list of networks and network sets that are available to a server profile along with their respective ports
+print("\nList all Ethernet networks associated with a server hardware type and enclosure group")
+available_networks = server_profiles.get_available_networks(
+    enclosureGroupUri=enclosure_group.data["uri"],
+    serverHardwareTypeUri=hardware_type.data["uri"], view='Ethernet')
+pprint(available_networks)
+
+# Get the all Ethernet networks associated with a server hardware type, enclosure group and scopeuris
+if oneview_client.api_version >= 600:
```
Comment (AnushaY1916) [inline examples/server_profiles.py]: This endpoint is supported in all the versions, may I know why this condition is kept here?

File: hpOneView/resources/servers/server_profiles.py [hunk 5]
```diff
@@ -46,86 +48,62 @@ class ServerProfiles(object):
...
-    def __init__(self, con):
-        self._connection = con
-        self._client = ResourceClient(con, self.URI)
+    def __init__(self, connection, data=None):
+        super(ServerProfiles, self).__init__(connection, data)
 
-    def create(self, resource, timeout=-1, force=''):
-        """
-        Creates a server profile using the information provided in the resource parameter.
+    def create(self, data=None, timeout=-1, force=''):
```
Comment (dileepds) [inline hpOneView/resources/servers/server_profiles.py]: better to use force is a Boolean flag with default value false

File: examples/server_profiles.py [hunk 6]
```diff
@@ -24,234 +24,216 @@
...
+profile.patch(operation="replace",
+              path="/templateCompliance", value="Compliant")
+pprint(profile.data)
 
 if oneview_client.api_version <= 500:
     # Retrieve ServerProfile schema
     # This method available only for API version <= 500
     print("\nRetrieve the generated ServerProfile schema")
-    schema = oneview_client.server_profiles.get_schema()
+    schema = profile.get_schema()
```
[the remark this answers is not in the document]
Comment (sijeesh) [reply examples/server_profiles.py]: Actually, we don't have to support schema endpoints in the resources. its got deprecated in almost all the resources and it's not consistent.

File: examples/server_profiles.py (hunk 4 above)
Comment (sijeesh) [reply examples/server_profiles.py]: This may be because of the query parameters, scope_uris param is got added in API version 600.

Comment (sijeesh) [reply examples/server_profiles.py]: Maybe I should update the comment here.

File: examples/server_profiles.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (sijeesh) [reply examples/server_profiles.py]: Remove this method from example file.

File: hpOneView/resources/servers/server_profiles.py (hunk 3 above)
Comment (sijeesh) [reply hpOneView/resources/servers/server_profiles.py]: Added

File: examples/server_profiles.py (hunk 4 above)
Comment (sijeesh) [reply examples/server_profiles.py]: Updated the comment

File: hpOneView/resources/servers/server_profiles.py (hunk 2 above)
Comment (sijeesh) [reply hpOneView/resources/servers/server_profiles.py]: Added

**Comment by sijeesh** [comment]
@soodpr , @AnushaY1916 , worked on all the comments. Please have a look.

**Pull request merged**